pkg/odo/cli/component/exec.go

```diff
@@ -64,7 +64,7 @@ Please provide a command to execute, odo exec -- <command to be execute>`)
 	eo.devfilePath = filepath.Join(eo.componentContext, devFile)
 
 	// if experimental mode is enabled and devfile is present
-	if experimental.IsExperimentalModeEnabled() {
+	if util.CheckPathExists(eo.devfilePath) {
```

Comment:
we should inform the user that this command doesn't work with s2i as the error message might be confusing
```
[mrinaldas@localhost ~]$ odo exec --context ~/project  -- ls 
 ✗  open /home/mrinaldas/project/devfile.yaml: no such file or directory
```

Reply:
Ah, you're right. I will add a test case for this.